crates/edr_evm/src/chain_spec.rs

```diff
@@ -26,40 +26,34 @@ use crate::{
 #[allow(clippy::trait_duplication_in_bounds)]
 pub trait ChainSpec:
     alloy_rlp::Encodable
+    // Defines the chain's internal types like blocks/headers or transactions
     + EthHeaderConstants
     + EvmWiring<
         Block: BlockEnvConstructor<Self, block::Header> + BlockEnvConstructor<Self, PartialHeader>,
         Transaction: alloy_rlp::Encodable
-                         + Clone
-                         + Debug
-                         + PartialEq
-                         + Eq
-                         + ExecutableTransaction
-                         + Transaction
-                         + TransactionType
-                         + TryFrom<
-            <Self as RpcSpec>::RpcTransaction,
-            Error = Self::RpcTransactionConversionError,
-        >,
-    > + RpcSpec<
-        ExecutionReceipt<FilterLog>: Debug,
+          + Clone
+          + Debug
+          + PartialEq
+          + Eq
+          + ExecutableTransaction
+          + Transaction
+          + TransactionType,
+    >
+    // Defines an RPC spec and conversion between RPC <-> EVM types
+    + RpcSpec<
         RpcBlock<<Self as RpcSpec>::RpcTransaction>: EthRpcBlock
-                                                         + TryInto<
-            EthBlockData<Self>,
-            Error = Self::RpcBlockConversionError,
-        >,
-        RpcReceipt: Debug
-                        + RpcTypeFrom<BlockReceipt<Self>, Hardfork = Self::Hardfork>
-                        + TryInto<BlockReceipt<Self>, Error = Self::RpcReceiptConversionError>,
+          + TryInto<EthBlockData<Self>, Error = Self::RpcBlockConversionError>,
         RpcTransaction: EthRpcTransaction
-                            + RpcTypeFrom<TransactionAndBlock<Self>, Hardfork = Self::Hardfork>,
-    > + RpcSpec<
+          + RpcTypeFrom<TransactionAndBlock<Self>, Hardfork = Self::Hardfork>
+          + TryInto<<Self as revm::primitives::EvmWiring>::Transaction, Error = Self::RpcTransactionConversionError>,
+        RpcReceipt: Debug
```

Comment:
What's the reason for inverting `RpcReceipt` and `RpcTransaction`? With this change it's no longer following the trait's (alphabetic) order

Reply:
I did not know that we should follow the alphabetic order, I can revert if that's the case, but this reads slightly better for me, as this follows the order from smaller to bigger building blocks (block -> tx -> receipt) and it only adds to the existing order from the `EvmWiring`, so kind of retains the symmetry.

Reply:
In the Hardhat repo we had a de facto policy¹ I like a lot: if a formatting-related thing (or any opinion-prone subject) can't be automatically enforced (with a formatter or a linter, for example), then that thing is not modified nor discussed. If we apply that here, then we have two options:

1) Configure `rustfmt` to sort this kind of thing alphabetically
2) If that's not possible, keep it as it was originally written

Notice this _technically_ means that adding a new element here and not inserting it in alphabetically order would be fair game. There's always nuance though: if something is already sorted, it probably makes sense to keep it that way.

I'm not (yet) saying we should follow this policy in EDR. But, in my experience, it's a sensible thing to do and it ends up saving a lot time and mental energy.

¹ AFAIK this is not documented anywhere.

Reply:
I'm happy to follow that policy. Agost and I agreed to that in the past too. We leaving the heavy lifting to the linters and rustfmt, but otherwise leave the code as it was done by the implementer. Otherwise it will be an endless (subjective) discussion about preferences. That's why I was asking why the ordering is being changed in this case.

As for code I write, I follow alphabetical order in ADTs and follow that order when using traits or initializing instances of ADTs (unless moving a value).

Reply:
In that case, yes, let's continue to do that. I'll add a topic for the technical sync just in case we want to discuss this a bit further.

Reply:
In general, I agree with this de facto policy, especially deferring to tools whenever possible to avoid needless discussions.

What I also found helpful in my experience is the argument that code is read more than it's written and that it's more valuable to explicitly make crucial/common concepts (like `ChainSpec` here) easier to grok when there's no tool or automatism to fall back on (or sometimes even going against it if it's makes it particularly hard to reason about the resulting code; not this case, though).

While I don't have a very strong preference here, it's just that I was having a hard time deciphering what is required/defined here, so I tried to clear it up and show some symmetry even if it meant going against the existing order.

I restored the order in https://github.com/NomicFoundation/edr/pull/640/commits/aca0abb6b623856b3a0278c861ce5a2b9d540ab4, let me know if I should change anything here.